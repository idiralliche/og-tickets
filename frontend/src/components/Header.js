import React, { useState } from 'react';
import NavLinks from './NavLinks';
import { FaBars, FaTimes } from 'react-icons/fa';

const Header = () => {
  const [menuOpen, setMenuOpen] = useState(false);

  const toggleMenu = () => {
    setMenuOpen(!menuOpen);
  };

  const closeMenu = () => {
    setMenuOpen(false);
  };

  return (
    <header className="header">
      <div className="logo">
        <a href='/'>
          <img src={`${process.env.PUBLIC_URL}/og-tickets-w.svg`} alt="og-tickets logo" />
        </a>
      </div>

      <nav>
        <NavLinks />

        <div className="burger-menu" onClick={toggleMenu}>{menuOpen ? <FaTimes /> : <FaBars />}</div>
        <div className={`mobile-nav ${menuOpen ? 'open' : ''}`} onClick={closeMenu}>
          <NavLinks onLinkClick={toggleMenu} />
        </div>
      </nav>
    </header>
  );
};

export default Header;
