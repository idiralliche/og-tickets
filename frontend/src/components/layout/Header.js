import React, { useState } from 'react';
import NavLinks from './NavLinks';
import { FaBars, FaTimes } from 'react-icons/fa';
import { Link } from 'react-router-dom';

const Header = () => {
  const [menuOpen, setMenuOpen] = useState(false);

  const toggleMenu = () => {
    setMenuOpen(!menuOpen);
  };

  const closeMenu = () => {
    setMenuOpen(false);
  };

  return (
    <header className='header'>
      <div className='logo'>
        <Link to='/'>
          <img
            src={`${process.env.PUBLIC_URL}/og-tickets-w.svg`}
            alt='og-tickets logo'
          />
        </Link>
      </div>

      <nav>
        <NavLinks />

        <div className='burger-menu' onClick={toggleMenu}>
          {menuOpen ? <FaTimes /> : <FaBars />}
        </div>
        <div
          className={`mobile-nav ${menuOpen ? 'open' : ''}`}
          onClick={closeMenu}
        >
          <NavLinks onLinkClick={toggleMenu} />
        </div>
      </nav>
    </header>
  );
};

export default Header;
