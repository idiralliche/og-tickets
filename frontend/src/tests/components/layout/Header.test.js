import React from 'react';
import { render, screen } from '@testing-library/react';
import Header from '../../../components/layout/Header';
import { BrowserRouter } from 'react-router-dom';

/**
 * Test suite for the Header component.
 * @module HeaderTests
 * @description Verifies the rendering and functionality of the Header component.
 */
describe('Header component', () => {
  /**
   * @test {Header} basic rendering
   * @description Verifies that the Header component renders the logo and navigation links correctly.
   * It checks for the presence of the logo and ensures that all navigation links are rendered
   * with the correct href attributes.
   */
  test('renders logo and navigation links', () => {
    render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    );

    // Verify that the logo is rendered (based on alt attribute text content, in the logo img tag)
    expect(screen.getByAltText(/og-tickets logo/i)).toBeInTheDocument();

    // Verify that navigation links from NavLinks are rendered
    const eventsLinks = screen.getAllByTestId('nav-link-events');
    const offersLinks = screen.getAllByTestId('nav-link-offers');
    const cartLinks = screen.getAllByTestId('nav-link-cart');
    const loginLinks = screen.getAllByTestId('nav-link-login');

    expect(eventsLinks).toHaveLength(2);
    expect(offersLinks).toHaveLength(2);
    expect(cartLinks).toHaveLength(2);
    expect(loginLinks).toHaveLength(2);

    eventsLinks.forEach((link) =>
      expect(link).toHaveAttribute('href', '/epreuves')
    );
    offersLinks.forEach((link) =>
      expect(link).toHaveAttribute('href', '/offres')
    );
    cartLinks.forEach((link) =>
      expect(link).toHaveAttribute('href', '/panier')
    );
    loginLinks.forEach((link) =>
      expect(link).toHaveAttribute('href', '/acces')
    );
  });

  /**
   * @test {Header} snapshot
   * @description Ensures the Header component matches its saved snapshot.
   */
  test('matches snapshot', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
