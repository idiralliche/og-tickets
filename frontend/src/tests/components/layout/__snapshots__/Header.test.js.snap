// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="header"
  >
    <div
      class="logo"
    >
      <a
        href="/"
      >
        <img
          alt="og-tickets logo"
          src="/og-tickets-w.svg"
        />
      </a>
    </div>
    <nav>
      <ul
        class="nav-links"
      >
        <li>
          <a
            data-discover="true"
            data-testid="nav-link-events"
            href="/epreuves"
          >
            Les épreuves
          </a>
        </li>
        <li>
          <a
            data-discover="true"
            data-testid="nav-link-offers"
            href="/offres"
          >
            Nos offres
          </a>
        </li>
      </ul>
      <ul
        class="nav-links"
      >
        <li>
          <a
            class="iconlink"
            data-discover="true"
            data-testid="nav-link-cart"
            href="/panier"
          >
            <span
              class="nav-link-icon cart"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 448 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M352 160v-32C352 57.42 294.579 0 224 0 153.42 0 96 57.42 96 128v32H0v272c0 44.183 35.817 80 80 80h288c44.183 0 80-35.817 80-80V160h-96zm-192-32c0-35.29 28.71-64 64-64s64 28.71 64 64v32H160v-32zm160 120c-13.255 0-24-10.745-24-24s10.745-24 24-24 24 10.745 24 24-10.745 24-24 24zm-192 0c-13.255 0-24-10.745-24-24s10.745-24 24-24 24 10.745 24 24-10.745 24-24 24z"
                />
              </svg>
            </span>
            <span
              class="nav-link-text"
            >
              Panier
            </span>
          </a>
        </li>
        <li>
          <a
            class="iconlink"
            data-discover="true"
            data-testid="nav-link-login"
            href="/acces"
          >
            <span
              class="nav-link-icon user"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 384 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M120 72c0-39.765 32.235-72 72-72s72 32.235 72 72c0 39.764-32.235 72-72 72s-72-32.236-72-72zm254.627 1.373c-12.496-12.497-32.758-12.497-45.254 0L242.745 160H141.254L54.627 73.373c-12.496-12.497-32.758-12.497-45.254 0-12.497 12.497-12.497 32.758 0 45.255L104 213.254V480c0 17.673 14.327 32 32 32h16c17.673 0 32-14.327 32-32V368h16v112c0 17.673 14.327 32 32 32h16c17.673 0 32-14.327 32-32V213.254l94.627-94.627c12.497-12.497 12.497-32.757 0-45.254z"
                />
              </svg>
            </span>
            <span
              class="nav-link-text"
            >
              Connexion
            </span>
          </a>
        </li>
      </ul>
      <div
        class="burger-menu"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
          />
        </svg>
      </div>
      <div
        class="mobile-nav "
      >
        <ul
          class="nav-links"
        >
          <li>
            <a
              data-discover="true"
              data-testid="nav-link-events"
              href="/epreuves"
            >
              Les épreuves
            </a>
          </li>
          <li>
            <a
              data-discover="true"
              data-testid="nav-link-offers"
              href="/offres"
            >
              Nos offres
            </a>
          </li>
        </ul>
        <ul
          class="nav-links"
        >
          <li>
            <a
              class="iconlink"
              data-discover="true"
              data-testid="nav-link-cart"
              href="/panier"
            >
              <span
                class="nav-link-icon cart"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 448 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M352 160v-32C352 57.42 294.579 0 224 0 153.42 0 96 57.42 96 128v32H0v272c0 44.183 35.817 80 80 80h288c44.183 0 80-35.817 80-80V160h-96zm-192-32c0-35.29 28.71-64 64-64s64 28.71 64 64v32H160v-32zm160 120c-13.255 0-24-10.745-24-24s10.745-24 24-24 24 10.745 24 24-10.745 24-24 24zm-192 0c-13.255 0-24-10.745-24-24s10.745-24 24-24 24 10.745 24 24-10.745 24-24 24z"
                  />
                </svg>
              </span>
              <span
                class="nav-link-text"
              >
                Panier
              </span>
            </a>
          </li>
          <li>
            <a
              class="iconlink"
              data-discover="true"
              data-testid="nav-link-login"
              href="/acces"
            >
              <span
                class="nav-link-icon user"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 384 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M120 72c0-39.765 32.235-72 72-72s72 32.235 72 72c0 39.764-32.235 72-72 72s-72-32.236-72-72zm254.627 1.373c-12.496-12.497-32.758-12.497-45.254 0L242.745 160H141.254L54.627 73.373c-12.496-12.497-32.758-12.497-45.254 0-12.497 12.497-12.497 32.758 0 45.255L104 213.254V480c0 17.673 14.327 32 32 32h16c17.673 0 32-14.327 32-32V368h16v112c0 17.673 14.327 32 32 32h16c17.673 0 32-14.327 32-32V213.254l94.627-94.627c12.497-12.497 12.497-32.757 0-45.254z"
                  />
                </svg>
              </span>
              <span
                class="nav-link-text"
              >
                Connexion
              </span>
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </header>
</DocumentFragment>
`;
